### PR TITLE
Add dummy fleet implementation, re-work update tests to use it

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/184

This is boyscouting to help with https://github.com/giantswarm/inago/issues/80, in that we can now set up and test the state of fleet much easier.

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

File: fleet/dummy.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		Config: DefaultDummyConfig(),
+		Units:  make(map[string]UnitStatus),
+	}, nil
+}
+
+// Submit creates a UnitStatus, and stores it.
+func (f *DummyFleet) Submit(ctx context.Context, name, content string) error {
+	f.Config.Logger.Debug(ctx, "dummy fleet: submit %v %v", name, content)
+
+	if f.Units == nil {
```
Comment (zeisss) [inline fleet/dummy.go]: Units is initalized in `NewDummyFleet`

File: fleet/dummy.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+type DummyConfig struct {
+	Logger logging.Logger
+}
+
+// DummyFleet is an implementation of the Fleet interface,
+// that is primarily intended to be used for testing.
+type DummyFleet struct {
+	Config DummyConfig
+	Units  map[string]UnitStatus
+}
```
Comment (zeisss) [inline fleet/dummy.go:22]: I would add a mutex

File: controller/update_test.go [hunk 3]
```diff
@@ -228,40 +210,65 @@ func TestIsGroupRemovalAllowed(t *testing.T) {
...
+						return strings.HasPrefix(s, "bluebird-unit@") && strings.HasSuffix(s, ".service")
+					},
+				)
+				if err != nil {
+					t.Fatal("Error returned getting statuses: ", err)
+				}
+				if len(unitStatusList) != 1 {
+					t.Fatal("Incorrect number of units:", len(f.Units))
+				}
+				if unitStatusList[0].Name == "bluebird-unit@1.service" {
```
Comment (zeisss) [inline controller/update_test.go]: :A: Do we want to use Gomega checks here? `Expect(unitStatusList[0]).To(Equal("bluebird-unit@1.service"))`
:B: Do we also verify the status is actually in started state?

File: fleet/dummy.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+func (f *DummyFleet) Submit(ctx context.Context, name, content string) error {
+	f.Config.Logger.Debug(ctx, "dummy fleet: submit %v %v", name, content)
+
+	f.Mutex.Lock()
+	defer f.Mutex.Unlock()
+
+	f.Units[name] = UnitStatus{
+		Current: unitStateLoaded,
+		Desired: unitStateLoaded,
+		Name:    name,
```
Comment (zeisss) [inline fleet/dummy.go:49]: No MachineStatus? boring :package:

**Comment by zeisss** [comment]
LGTM, although the dummy is really dumb and we might be testing the wrong thing.

File: controller/update_test.go (hunk 3 above)
Comment (JosephSalisbury) [inline controller/update_test.go]: a) I'm not a _massive_ fan of gomega, to be honest
b) we should do that, i'll add it

File: fleet/dummy.go (hunk 4 above)
Comment (JosephSalisbury) [inline fleet/dummy.go:49]: Didn't need it for now

**Pull request merged**